docs/adr/0006_component_constructors.md

```diff
@@ -0,0 +1,363 @@
+
+* Providing centralized access and interaction with `component-constructor.yaml` style files and their schemas
+* Providing centralized extension points for adding new Input Methods for building component versions in addition
+  to directly referencing an access.
+
+## Input Methods and Contracts
+
+A key feature of the `component-constructor.yaml` file is the ability to define **inputs**—specifications that describe where to retrieve data and how to store it in a resource for a newly created component version.
+
+### Purpose of Input Methods
+
+An **input method** replaces the `access` field in a standard OCM resource or source. When specified, the input is interpreted at runtime during component construction and used to generate an appropriate access specification based on the processed input data.
+
+### Example
+
+The following `component-constructor.yaml` defines a simple input method:
+
+```yaml
+components:
+  - name: github.com/acme.org/helloworld
+    version: 1.0.0
+    provider:
+      name: internal
+    resources:
+      - name: testdata
+        type: blob
+        relation: local
+        input:
+          type: file
+          path: ./testdata/text.txt
+```
+
+which could be processed by a constructor library as follows:
+
+```yaml
+component:
+  componentReferences: []
+  name: github.com/acme.org/helloworld
+  provider: internal
+  repositoryContexts: []
+  resources:
+  - access:
+      localReference: sha256:c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2
+      mediaType: application/octet-stream
+      type: localBlob/v1
+    creationTime: "2025-05-19T08:28:30+02:00"
+    digest:
+      hashAlgorithm: SHA-256
+      normalisationAlgorithm: genericBlobDigest/v1
+      value: c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2
+    name: testdata
+    relation: local
+    size: 6
+    type: blob
+    version: 1.0.0
+  sources: []
+  version: 1.0.0
+meta:
+  schemaVersion: v2
+
+``` 
+
+In this example:
+
+- The input of type `file` tells the constructor to use the registered `file` input method.
+- This method processes `./testdata/text.txt` and generates an access specification.
+- The result is included in the final component descriptor.
+- The format of the generated access specification depends on the specific input method used.
+
+---
+
+### Categories of Input Methods
+
+Input methods are typically categorized into two types:
+
+#### Local Input Methods
+
+- Store the content as a `localBlob` next to the generated component version.
+
+#### Global Input Methods
+
+- Generate access specifications that are independent of the constructed component version.
+
+Each input method must determine whether it should produce a `localBlob` or a global access specification.
+
+---
+
+### "By Value" Construction
+
+Any input or access specification can be marked as **constructed by value**. In this case:
+
+- The constructor directly embeds the resource into the component version.
```

Comment:
And here you don't mean that the actual byte stream is added to the component version yaml file. Just to be 100% clear on that, right?

Reply:
The component version contains a descriptor which receives this resource through the embedding. Does that make more sense?

Reply:
Yep